src/ContainerRenameModal.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import {
+    Button,
+    Form, FormGroup,
+    Modal, TextInput
+} from '@patternfly/react-core';
+import cockpit from 'cockpit';
+
+import * as client from './client.js';
+
+import { ErrorNotification } from './Notification.jsx';
+import * as dockerNames from 'docker-names';
+
+const _ = cockpit.gettext;
+
+class ContainerRenameModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: dockerNames.getRandomName(),
+            nameError: "",
+        };
+
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleRename = this.handleRename.bind(this);
+    }
+
+    componentDidMount() {
+        this._isMounted = true;
+    }
+
+    componentWillUnmount() {
+        this._isMounted = false;
+
+        if (this.activeConnection)
+            this.activeConnection.close();
+    }
+
+    handleInputChange(targetName, value) {
+        const newState = { [targetName]: value };
+
+        if (targetName === "name")
+            newState.nameError = "";
+
+        this.setState(newState);
+    }
+
+    handleRename() {
+        if (!this.state.name) {
+            this.setState({ nameError: "Container name is required" });
```

Comment:
Missing translations `_("")`

Reply:
Thanks, fixed in commit https://github.com/cockpit-project/cockpit-podman/commit/0c3706e1009fabc1fee2a17b247175993469a9c4